### DISCUSSION TITLE
Is there any way to support resources that can only be replaced and not updated in place?

### DISCUSSION THREAD

**Discussion by empath-nirvana** [crossplane/upjet#286] [Q&A]
I'm trying to create a github organization secret with a custom github provider i made with upjet.

Currently, the terraform provider for github marks all fields for a github organization secret as "forceNew", so we can't update the secret.

There seems to be no way to allow prevent_destroy on any resource?  Is there any way to make this work without deleting the crossplane resource and recreating it?

**Comment by bradkwadsworth-mw** [comment]
+1
I've run into this issue when trying to re-create IAM members in GCP if a role or member needs to be modified.

**Comment by jeanduplessis** [comment]
Would [this proposal](https://github.com/crossplane/crossplane/blob/master/design/one-pager-ignore-changes.md) that introduces more granular management policies help here?

**Comment by ulucinar** [comment]
Please also see the "Note" here:
https://docs.crossplane.io/latest/concepts/managed-resources/#immutable-fields

By convention, Crossplane providers never delete resources based on a `spec.forProvider` change. I think what's suggested here deserves a discussion in the upstream Crossplane community.

**Comment by malins** [comment]
I have the same problem with a terraform resource which marks most fields with "ForceNew: true" (https://github.com/bpg/terraform-provider-proxmox/blob/main/proxmoxtf/resource/file.go#L77). It would be nice that we could override the "prevent-destroy" TF lifecycle attribute for such resources as part of Upjet configuration.

I cannot "fix" this even by setting managementPolicies to [Create, Delete, Observe], since the error occurs in observe.

_cannot run plan: plan failed: Instance cannot be destroyed: Resource proxmox_virtual_environment_file.cloud-init-test has lifecycle.prevent_destroy set, but the plan calls for this resource to be destroyed. To avoid this error and continue with the plan, either disable lifecycle.prevent_destroy or reduce the scope of the plan using the -target option._
